internal/handlers/common/update_array_operators.go

```diff
@@ -414,3 +415,66 @@ func processPullAllArrayUpdateExpression(doc, update *types.Document) (bool, err
 
 	return changed, nil
 }
+
+// processPullArrayUpdateExpression changes document according to $pull array update operator.
+// If the document was changed it returns true.
+func processPullArrayUpdateExpression(doc *types.Document, update *types.Document) (bool, error) {
+	var changed bool
+
+	iter := update.Iterator()
+
+	for {
+		key, pullValueRaw, err := iter.Next()
+		if err != nil {
+			if errors.Is(err, iterator.ErrIteratorDone) {
+				break
+			}
+
+			return false, lazyerrors.Error(err)
+		}
+
+		path, err := types.NewPathFromString(key)
+		if err != nil {
+			return false, lazyerrors.Error(err)
+		}
+
+		// If the path does not exist, create a new array and set it.
```

Comment:
This is the part I don't understand: if the path doesn't exist, we create an empty array and then try to pull from it? But then we will not pull anything, and `changed` will be `false`, right? Then does it make sense to create this empty array at all? Can't we just return `false, nil` right here?

Reply:
Yeah creating an array here but without setting changed = `true` is confusing. Given codecov says it has tested it, compat must also creates and empty array without setting changed = `true`.

Reply:
I've added a comment that possibly could help to understand why that `SetByPath` call is needed.